knowledge-base/pdfviewer-sign-pdfs.md

```diff
@@ -0,0 +1,152 @@
+		x509Store.Open(OpenFlags.ReadOnly | OpenFlags.OpenExistingOnly);
+		var validCerts = x509Store.Certificates.Find(X509FindType.FindByTimeValid, DateTime.Now, false);
+		
+		// IMPORTANT: This example is just taking the very first valid cert. In a real app, you will selected the cert you want to sign the document with
+		certificate = validCerts.FirstOrDefault();
+
+
+        // **** PHASE 2 - Add certificate to SignatureField and use certificate **** //
+
+        // The name of the signature must be unique.
+        var signatureName = "John Doe";
+
+        // The Signature object is added to a signature field, so we can add a visualization to it.
+		var signatureField = new SignatureField(signatureName);
+		signatureField.Signature = new Telerik.Windows.Documents.Fixed.Model.DigitalSignatures.Signature(certificate);
+
+        // close the local cert store now that we're done
+        x509Store.Close();
+
+
+        // **** PHASE 3 - IMPORT DOCUMENT and insert signature field **** //
+
+        var document = new PdfFormatProvider().Import(unsignedDocumentBytes);
+
+        // For the sake of this demo, we are adding a new page and adding a new signature field there, if your document already has a signature field, you can search for it instead
+        RadFixedPage page = document.Pages.AddPage();
+
+        // This is the Form XObject element that represents the contents of the signature field.
+        var form = new Telerik.Windows.Documents.Fixed.Model.Objects.Form
+        {
+            FormSource = new FormSource { Size = new Size(220, 220) }
+        };
+
+        // We will use the editor to fill the Form XObject.
+        var formEditor = new FixedContentEditor(form.FormSource);
+        form.Position.Translate(10, 10);
+        formEditor.DrawCircle(new Point(50, 50), 20);
+        formEditor.DrawText(signatureName);
+
+		// The widget contains the Form XObject and defines the appearance of the signature field.
+		var widget = signatureField.Widgets.AddWidget();
+		widget.Rect = new Rect(200, 600, 100, 100);
+		widget.Border = new AnnotationBorder(10, AnnotationBorderStyle.Solid, null);
+		widget.Content.NormalContentSource = form.FormSource;
+		widget.RecalculateContent();
+
+        // Finally add the SignatureWidget to the page's annotations
+		page.Annotations.Add(widget);
+
+		var editor = new FixedContentEditor(page);
+		editor.Position.Translate(200, 400);
+		editor.DrawForm(form.FormSource);
+		document.AcroForm.FormFields.Add(signatureField);
+		widget.RecalculateContent();
+		widget.AppearanceCharacteristics.Background = new Telerik.Windows.Documents.Fixed.Model.ColorSpaces.RgbColor(255, 0, 0);
+
+        // **** PHASE 4 - EXPORT DOCUMENT **** //
+        byte[] signedDocBytes = new PdfFormatProvider().Export(document);
+
+        return signedDocBytes;
+    }
+}
+```
+
+## Notes
+
+- The Telerik Blazor PdfViewer does not currently support digital signature management in the UI.
```

Comment:
This repeats similar content above. Since it's the reason to create a KB, I guess the statement is unnecessary here.

Reply:
Handled